### DISCUSSION TITLE
How to Prevent Hitting Vercel Free Limits

### DISCUSSION THREAD

**Discussion by ilovefreesw** [walinejs/waline#1918] [General]
Hi, today i got this message from Vercel.

![image](https://user-images.githubusercontent.com/77317006/229411069-c3336d71-1518-4d23-942f-de86be39170a.png)

Is there way to host it on Vercel and not go over the free limits. The bandwidth is under limit but the execution has exceeded 100GB-HR.

**Comment by lizheming** [comment]
Can you confirm whether your website traffic is normal? In most cases, you should not reach this limit.
If your website PV is really high, you can pay to use Vercel, but it may not be very cost-effective to host on Vercel. Consider using [Deta](https://waline.js.org/en/guide/deploy/deta.html) deployment or self-hosting to circumvent this problem.

**Comment by ilovefreesw** [comment]
Yes I get around 200000+ views on my website. But very less comments. Like 100 comments in a month perhaps.

**Reply by lizheming** [reply]
If this is the case, it is normal to reach the limit, because the display of the comment list requires a request interface. I think you can have the following solutions to solve this problem:
1. Consider lazy loading Waline. Not all users who enter the page will read the comments, so this should reduce the magnitude of the requests.
2. Migrate the Waline server to Deta, which has no traffic limit. But I'm not sure if there will be other restrictions, after all, it's a free service, we can't ask too much.
3. Considering that your page comments read more and write less, you can consider calling Waline's API to directly inject comment data into the page through the SSG solution. This minimizes your API request traffic.

**Reply by ilovefreesw** [reply]
Okay let's see what I can do.

**Reply by ilovefreesw** [reply]
![image](https://user-images.githubusercontent.com/77317006/229800644-d1b4b1f0-904a-4ef8-9064-6e3cb76b2628.png)

I discovered the problem the timed out functions taking most of the allowance 🤔

**Reply by lizheming** [reply]
You can go to the log page to see which specific api timed out and why they timed out

**Reply by ilovefreesw** [reply]
I think it is this:

![image](https://user-images.githubusercontent.com/77317006/229822857-980e031a-992c-4a73-99b1-a3e970bed1e0.png)

**Reply by lizheming** [reply]
This should be normal, it should be because you set SECURE_DOMAINS and then other websites use your service to return an error
